src/container/srv_container.c

```diff
@@ -1383,6 +1383,9 @@ cont_agg_eph_leader_ult(void *arg)
 	struct cont_ec_agg	*tmp;
 	int			rc = 0;
 
+	if (svc->cs_ec_leader_ephs_req == NULL)
```

Comment:
Why does this become needed ?

Reply:
@bfaccini, these changes correspond to the following in the commit message:

> Also, in some error paths among those cases, the ULTs may not terminate, leading to hangs. This patch adopts a check-if-req-is-null-in-the-ULT approach from other error paths, whose correctness subtly relies on the absence of yielding between the ULT creation and the ABT_thread_free call---something that should be replaced with a more robust solution in the future.

Does this answer your question?

Reply:
Well, in fact I should have already added that this description was a bit obscure to me ...

Reply:
@bfaccini, if cont_svc_ec_agg_leader_start fails to get a sched_req and attempts to handle the error by calling ABT_thread_free to join and free ec_eph_leader_ult, it doesn't do anything to make sure the ULT will actually abort and terminate. Niu and I think the sched_req_get API should be revised to hide this error handling from callers. That is, however, not something we feel comfortable to do for 2.0. This patch therefore adopts the check-if-req-is-null-in-the-ULT approach used by dtx to address the issue.

Reply:
hmm, ok understood. And I presume this is for the same check-if-req-is-null-in-the-ULT approach that @NiuYawei thinks that to simply fix DAOS-8826/PR-7021 I should just "change sched_req_wakeup() to check passed in 'req' and return if 'req == NULL'" , right ?
But am I wrong to think that this check-if-req-is-null-in-the-ULT approach is a bit racy ??

Reply:
@bfaccini, it _is_ racy in the sense that "[its] correctness subtly relies on the absence of yielding between the ULT creation and the ABT_thread_free call", which is true in current code. That's why we feel a more robust solution is required.